Sources/GoogleAI/FunctionCalling.swift

```diff
@@ -177,6 +177,40 @@ public struct Tool: Encodable {
   }
 }
 
+public enum Mode: String, Encodable {
```

Comment:
Drive by review: `Mode` is too generic for a public API here, should be something like `FunctionCallingMode`

Reply:
Done. Thanks!

Reply:
Paul already updated this to `FunctionCallingMode` but wanted to suggest another alternative of nesting the `Mode` enum in `FunctionCallingConfig` (similar to what we did for `ModelContent.Part`). This would be closer to the proto definition (and the Android implementation in https://github.com/google/generative-ai-android/pull/103) but I don't feel strongly about one approach or the other as long as we're consistent.